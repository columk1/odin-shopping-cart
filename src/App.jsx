import './index.css'
import { createBrowserRouter, RouterProvider } from 'react-router-dom'
import Root from './routes/root'
import ErrorPage from './ErrorPage.jsx'
import Hero from './components/Hero/Hero'
import Shop, { loader as shopLoader } from './routes/shop.jsx'
import Products from './components/Products/Products'
import Product from './components/Product/Product'
import Cart from './routes/cart/Cart.jsx'

const router = createBrowserRouter([
  {
    path: '/',
    element: <Root />,
    errorElement: <ErrorPage />,
    children: [
      { index: true, element: <Hero /> },
      {
        path: 'shop',
        element: <Shop />,
        loader: shopLoader,
        children: [
          { index: true, element: <Products /> },
          { path: ':productId', element: <Product /> },
        ],
      },
      {
        path: 'cart',
        element: <Cart />,
      },
    ],
  },
])

export default function App() {
  return <RouterProvider router={router} />
}
