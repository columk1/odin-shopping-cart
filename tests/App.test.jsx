import { describe, it, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import App from '../src/App.jsx'
import { userEvent } from '@testing-library/user-event'

describe('App', async () => {
  it('Renders the homepage (Root.jsx with Hero.jsx in the outlet)', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })

  it('Renders a header element', () => {
    render(<App />)
    const header = screen.getByRole('banner')
    expect(header).toBeInTheDocument()
  })
  it('Renders a navbar', () => {
    render(<App />)
    const nav = screen.getByRole('navigation')
    expect(nav).toBeInTheDocument()
  })
})

describe('Routes configured correctly', () => {
  it('Renders homepage', async () => {
    render(<App />)

    expect(screen.getByRole('heading', { level: 2 }).textContent).toMatch('A revolution begins')
    expect(screen.getByRole('button').textContent).toMatch('Start Shopping')
  })

  //   it('Renders shop page', async () => {
  //     const user = userEvent.setup()
  //     render(<App />)
  //     await user.click(screen.getByRole('link', { name: 'shopping_cart' }))
  //     screen.debug()
  //   })
})
