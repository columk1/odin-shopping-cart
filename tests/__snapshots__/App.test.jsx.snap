// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Renders the homepage (Root.jsx with Hero.jsx in the outlet) 1`] = `
<div>
  <header
    class="_header_63c2ea"
  >
    <h1
      class="_logo_63c2ea"
    >
      StuffShop
    </h1>
    <nav
      id="mainNav"
    >
      <ul
        class="_ul_214497"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            href="/shop"
          >
            Shop
          </a>
        </li>
        <li>
          <a
            class=""
            href="/cart"
          >
            <span
              class="material-symbols-outlined"
            >
              shopping_cart
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    id="detail"
  >
    <section
      class="_hero_84d173"
    >
      <div
        class="_heroContent_84d173"
      >
        <h2
          class="_h1_84d173"
        >
          A revolution begins with new shoes
        </h2>
        <p>
          This website has some subtext that goes here under the main title. It's a smaller font and it says something about shoes.
        </p>
        <a
          class=""
          href="/shop"
        >
          <button>
            Start Shopping
          </button>
        </a>
      </div>
      <div
        class="_heroImg_84d173"
      >
        <span
          class="material-symbols-outlined"
        >
          favorite
        </span>
      </div>
    </section>
  </div>
</div>
`;
